locales/i18n.js

```diff
@@ -19,11 +19,20 @@ function formatLocale(lang) {
 function urlOverrideLang(path) {
   var localPath = path || location.pathname;
   var localeCode = localPath.split('/')[1];
-  var pathname = localPath.split('/')[2] || localeCode;
+  localeCode = (/^[a-z]{2,3}(-[A-Z]{2})?$/.test(localeCode)) ? localeCode : null;
```

Comment:
Cool cool, works for me.

I've been meaning to move some of this to the server side and grab the locale from the headers, but havn't had a chance. At that point, we can reconsider the viability of a langmap.